lib/browser/api/web-request.ts

```diff
@@ -0,0 +1,62 @@
+import * as deprecate from '@electron/internal/common/deprecate';
+
+const { WebRequest } = process._linkedBinding('electron_browser_web_request');
+
+const urlsDeprecated = deprecate.warnOnce('Empty url array in WebRequestFilter', '<all_urls> to match all URLs');
+function modifyArgs (args: any[]) {
+  if (args.length < 2) return;
+  const filter = args[0];
+
+  if (filter.urls.length === 0) {
+    urlsDeprecated();
+    filter.urls = ['<all_urls>'];
+  }
+}
```

Comment:
We should consider moving this into C++ if possible.

Here's an example usage of `electron::utils::EmitWarning` for how we can do deprecation warnings there

https://github.com/electron/electron/blob/493e3c4537bfa70405a2da998503e82594eada14/shell/browser/native_window_mac.mm#L182-L184

Reply:
ohhh this would make my implementation a lot simpler if I add the deprecation message to C++. I will try this out!

Reply:
Just for my own learning, what's the rule of thumb when you do the modification in C++ vs Typescript wrapper?

Reply:
I'd say in this case, I'm making the recommendation as it's easier to maintain. Less code spread over less files. It's also more efficient as there's less JavaScript to parse.

I think we tend to write deprecations in JS if existing JS code exists for the class.

Reply:
Thanks! Makes sense! I was able to delete a bunch of JS code! New deprecation message!
![image](https://github.com/user-attachments/assets/3cdc0fc4-c28c-46e7-a6f5-fa1a489a0b34)